### ISSUE TITLE
add mpi4py package?

### ISSUE THREAD

**Issue by francesco-ballarin** [pyodide/pyodide#4149]
## 🐍 Package Request

- Package Name and Version: mpi4py/latest
- Package URL: https://github.com/mpi4py/mpi4py
- Package Dependencies that needs to be resolved first: `mpich` or `openmpi`

Hi,
has anyone ever tried to add `mpi4py` to `pyodide`?
I can't find any mention of `MPI` in the repository, except for the `h5py` package, in which `MPI` is explicitly disabled.

The challenge is surely to get either `mpich` or `openmpi` to compile. I tried it myself, and was not even able to get either of them to successfully configure with `emconfigure`.

I do realize that the question may sound odd to many jupyter users, in the sense that jupyter notebooks are typically run in serial rather than in parallel. My motivation for asking is that many of the libraries I use (see e.g. https://fem-on-colab.github.io/packages.html) have a hard dependency on `MPI` and `mpi4py`, even when the code is actually run in serial (i.e., the code will be run on an MPI communicator with size 1).

Thanks,
Francesco

**Comment by rth** [comment]
Mpi is a network protocol. In the browser, you cannot open unix socket so it will generally not work. See [emscripten docs](https://emscripten.org/docs/porting/networking.html#emulated-posix-tcp-sockets-over-websockets)  for possible workarounds.  In any case, you would need to run a proxy to translate the connections done by the browser to MPI connections somehow.

This is currently out of scope for Pyodide, but you are welcome to experiment with it.

**Comment by francesco-ballarin** [comment]
Thanks @rth for your answer, and for providing the link to the emscripten docs. I am satisfied with the content there, and I am going to close this since you told me that this out of scope for the project.
